shell/browser/native_window_mac.mm

```diff
@@ -473,6 +473,12 @@ void ViewDidMoveToSuperview(NSView* self, SEL _cmd) {
     return;
   }
 
+  // If a sheet is attached to the window when we call
+  // [window_ performClose:nil], the window won't close properly
+  // even after the user has ended the sheet.
+  // Ensure it's closed before calling [window_ performClose:nil].
+  SetEnabled(true);
```

Comment:
The `Close()` would then be able to close an explicitly disabled window, but it seems that `SetEnabled(false)` crashes on macOS, so it probably does not matter.

Reply:
that was essentially my thought process as well - happy to consider another approach but this seemed to be the best all-around solution 🤔